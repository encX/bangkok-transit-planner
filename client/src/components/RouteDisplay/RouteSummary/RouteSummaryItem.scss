@import "src/css/Common";

.route-summary-item {
	&--name {
		border: 1px solid $black-normal;
		padding: 0 4px;
		text-align: center;
	}

	&--flag {
		background-color: $black-normal;
		color: #FFFFFF;
		padding: 1px 4px;
		text-align: center;
	}

	&--value {
		padding: 1px;
		font-weight: bold;
	}
}
