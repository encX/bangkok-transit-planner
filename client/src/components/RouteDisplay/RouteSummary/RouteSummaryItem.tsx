import React, {FC} from 'react';
import {Row, Col} from 'antd';

import './RouteSummaryItem.scss';

interface RouteSummaryItemProps {
	name: string;
	value: string;
	flag?: string;
	color?: string;
}

export const RouteSummaryItem: FC<RouteSummaryItemProps> = ({name, value, flag, color}): JSX.Element => {
	return (
		<Row gutter={[16, 8]}>
			<Col flex="100px"><div className="route-summary-item--name" style={{borderColor: color}}>{name}</div></Col>
			<Col flex="1"><div className="route-summary-item--value">{value}</div></Col>
			{flag && <Col flex="100px"><div className="route-summary-item--flag" style={{backgroundColor: color}}>{flag}</div></Col>}
		</Row>
	);
};
