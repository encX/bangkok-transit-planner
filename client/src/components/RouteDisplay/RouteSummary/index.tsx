import React, {FC} from 'react';
import {RouteSummaryItem} from './RouteSummaryItem';

interface RouteSummaryProps {
	cost: number;
	time: string;
	transfer: number;
}

const RouteSummary: FC<RouteSummaryProps> = ({cost, time, transfer}): JSX.Element => {
	return (
		<>
			<RouteSummaryItem name="Fare" value={`${cost} Baht`}/>
			<RouteSummaryItem name="Time" value={time}/>
			<RouteSummaryItem name="Transfer" value={`${transfer} times`}/>
		</>
	);
};

export default RouteSummary;
