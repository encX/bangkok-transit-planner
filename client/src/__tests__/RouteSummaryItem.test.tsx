import React from 'react';
import {shallow} from 'enzyme';
import {RouteSummaryItem} from '../components/RouteDisplay/RouteSummary/RouteSummaryItem';


describe('RouteSummaryItem', function() {
	it('only required props: renders correctly', () => {
		const tree = shallow(<RouteSummaryItem name="name" value="value"/>);
		expect(tree).toMatchSnapshot();
	});
	
	it('with optional props: renders correctly', () => {
		const tree = shallow(<RouteSummaryItem name="name" value="value" flag="flag" color="#ABCABC"/>);
		expect(tree).toMatchSnapshot();
	});
});
