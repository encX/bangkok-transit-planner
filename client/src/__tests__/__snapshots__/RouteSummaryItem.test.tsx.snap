// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteSummaryItem only required props: renders correctly 1`] = `
<Row
  gutter={
    Array [
      16,
      8,
    ]
  }
>
  <Col
    flex="100px"
  >
    <div
      className="row-summary-item--name"
      style={
        Object {
          "borderColor": undefined,
        }
      }
    >
      name
    </div>
  </Col>
  <Col
    flex="1"
  >
    <div
      className="row-summary-item--value"
    >
      value
    </div>
  </Col>
</Row>
`;

exports[`RouteSummaryItem with optional props: renders correctly 1`] = `
<Row
  gutter={
    Array [
      16,
      8,
    ]
  }
>
  <Col
    flex="100px"
  >
    <div
      className="row-summary-item--name"
      style={
        Object {
          "borderColor": "#ABCABC",
        }
      }
    >
      name
    </div>
  </Col>
  <Col
    flex="1"
  >
    <div
      className="row-summary-item--value"
    >
      value
    </div>
  </Col>
  <Col
    flex="100px"
  >
    <div
      className="row-summary-item--flag"
      style={
        Object {
          "backgroundColor": "#ABCABC",
        }
      }
    >
      flag
    </div>
  </Col>
</Row>
`;
